processing/src/main/java/org/apache/druid/segment/virtual/ExpressionSelectors.java

```diff
@@ -381,8 +383,12 @@ public ExpressionType getType(String name)
         return selector.lookupName(row.get(0));
       } else {
         // column selector factories hate you and use [] and [null] interchangeably for nullish data
-        if (row.size() == 0) {
-          return new Object[]{null};
+        if (row.size() == 0 || (row.size() == 1 && selector.getObject() == null)) {
+          if (homogenize) {
```

Comment:
maybe I am overthinking but does it make sense to take out this branch and return a different supplier itself? Might be unnecessary if this method is not very hot.

Reply:
eh, it is a hot path, though probably not the worst thing happening in non-vectorized expressions, so an extra branch probably doesn't do _that_ much heh

that said, I think these input binding suppliers are probably worth thinking about either optimizing with additional specialized implementations, or at least using as an example of what not to do when we add multi-value string and array support for vectorized expressions.